crates/pathfinder/src/storage/contract.rs

```diff
@@ -64,10 +56,10 @@ impl ContractsTable {
             .context("Failed to compress definition")?;
 
         transaction.execute(
-            r"INSERT INTO contracts ( address,  hash, bytecode,   abi,  definition)
-                             VALUES (:address, :hash, :bytecode, :abi, :definition)",
+            r"INSERT INTO contract_code ( hash,  bytecode,  abi,  definition)
+                             VALUES (:hash, :bytecode, :abi, :definition)
+                             ON CONFLICT DO NOTHING",
```

Comment:
```suggestion
                             VALUES (:hash, :bytecode, :abi, :definition);
```
We should not be trying to insert duplicates -- this would be a bug for us I think. When inserting, we should instead first check if it exists and __only then__ go download the contract (if it doesn't exist already).

Reply:
Ahh yes, correct :+1: